### PR TITLE
[Runtimes] Fix - propagate databricks job errors

### PR DESCRIPTION
**Pull request by tomerm-iguazio**
1) Capture errors that occur within the `Databricks environment`, raise them to make the real error message accessible to the user.
2) Provide additional information such as the `trace`, `metadata`, and other relevant details.
3) Test that demonstrates this usage.

[ML-4403](https://jira.iguazeng.com/browse/ML-4403)

### CODE REVIEW

File: tests/system/runtimes/test_databricks.py [hunk 1]
```diff
@@ -118,6 +119,32 @@ def print_kwargs(**kwargs):
...
+        function = function_ref.to_function()
+
+        self._add_databricks_env(function=function, is_cluster_id_required=True)
+        with pytest.raises(RunError) as error:
+            function.run(
+                handler="import_mlrun",
+                project="databricks-proj",
+            )
+        lines = str(error.value).splitlines()
+        assert "No module named 'mlrun'" in lines[2]
```
Comment (gtopper) [inline tests/system/runtimes/test_databricks.py:145]: I ran it and it failed on
```
>           assert "No module named 'mlrun'" in lines[2]
E           IndexError: list index out of range
```

File: mlrun/runtimes/databricks/databricks_wrapper.py [hunk 2]
```diff
@@ -84,10 +86,21 @@ def print_status(run: Run):
...
-            timeout=datetime.timedelta(minutes=timeout_minutes),
-            callback=print_status,
-        )
+        try:
+            run = waiter.result(
+                timeout=datetime.timedelta(minutes=timeout_minutes),
+                callback=print_status,
+            )
+        except OperationFailed:
+            # TODO handle rerun tasks - so we can not take the first task in tasks list.
```
Comment (gtopper) [inline mlrun/runtimes/databricks/databricks_wrapper.py:95]: Why not deal with it now?

Comment (tomerm-iguazio) [reply mlrun/runtimes/databricks/databricks_wrapper.py:95]: I will open another PR to handle rerun issue:)

Comment (gtopper) [reply mlrun/runtimes/databricks/databricks_wrapper.py:95]: Why? This is new code. Might as well get it right the first time if there's no particular reason not to.

File: tests/system/runtimes/test_databricks.py (hunk 1 above)
Comment (tomerm-iguazio) [reply tests/system/runtimes/test_databricks.py:145]: Have you upgraded your server to this mlrun PR version?

Comment (gtopper) [reply tests/system/runtimes/test_databricks.py:145]: Got it.

**Review by gtopper — APPROVED** [review]
✅

**Pull request merged**